cloudinit/net/__init__.py

```diff
@@ -449,36 +500,39 @@ def find_fallback_nic_on_linux(blacklist_drivers=None):
             msg = "Waiting for udev events to settle"
             util.log_time(LOG.debug, msg, func=util.udevadm_settle)
 
-    # get list of interfaces that could have connections
-    invalid_interfaces = set(["lo"])
-    potential_interfaces = set(
-        [
-            device
-            for device in get_devicelist()
-            if device_driver(device) not in blacklist_drivers
-        ]
-    )
-    potential_interfaces = potential_interfaces.difference(invalid_interfaces)
     # sort into interfaces with carrier, interfaces which could have carrier,
     # and ignore interfaces that are definitely disconnected
     connected = []
     possibly_connected = []
-    for interface in potential_interfaces:
+    for interface in get_devicelist():
+        if interface == "lo":
+            continue
+        driver = device_driver(interface)
+        if driver in blacklist_drivers:
+            LOG.debug(
```

Comment:
I don't know if these debug logs are excessive, but I figured we should have some indication of why we are ignoring candidate NICs.